scanner/src/main/java/ca/on/oicr/gsi/runscanner/scanner/processor/RevioPacBioProcessor.java

```diff
@@ -0,0 +1,335 @@
+      String resultFolder = (String) expr[2].evaluate(document, XPathConstants.STRING);
+      String poolName = (String) expr[3].evaluate(document, XPathConstants.STRING);
+
+      // From DefaultPacBio checking to make sure string is valid
+      if (isStringBlankOrNull(position)
+          || isStringBlankOrNull(resultFolder)
+          || isStringBlankOrNull(containerSerialNumber)
+          || isStringBlankOrNull(poolName)) {
+        return;
+      }
+      // Container represents one SMRT Cell
+      Map<String, String> containerInfo = new HashMap<>();
+      containerInfo.put("ResultFolder", resultFolder);
+      containerInfo.put("Position", position);
+      containerInfo.put("ContainerSerialNumber", containerSerialNumber);
+      containerInfo.put("PoolName", poolName);
+
+      // Add container to SMRT cell positionList
+      List<Map<String, String>> tempPositionList = dto.getPositionList();
+      if (dto.getPositionList() == null) {
+        tempPositionList = new ArrayList<>();
+      }
+      tempPositionList.add(containerInfo);
+      dto.setPositionList(tempPositionList);
+    };
+  }
+
+  /**
+   * Extract a string expression from the metadata file and write it into the DTO.
+   *
+   * @param expression the XPath expression yielding the string
+   * @param setter writer for the string
+   * @return
+   */
+  private static RevioPacBioProcessor.ProcessMetadata processString(
+      String expression, BiConsumer<PacBioNotificationDto, String> setter) {
+    XPathExpression expr = RunProcessor.compileXPath(expression)[0];
+    return (document, dto, timeZone) -> {
+      String result = (String) expr.evaluate(document, XPathConstants.STRING);
+      if (result != null) {
+        setter.accept(dto, result);
+      }
+    };
+  }
+
+  public RevioPacBioProcessor(Builder builder, String s) {
+    super(builder);
+  }
+
+  public static RunProcessor create(Builder builder, ObjectNode parameters) {
+    JsonNode address = parameters.get("address");
+    return address.isTextual()
+        ? new RevioPacBioProcessor(builder, address.textValue().replaceAll("/+$", ""))
+        : null;
+  }
+
+  @Override
+  public Stream<File> getRunsFromRoot(File root) {
+    return Arrays.stream(
+        root.listFiles(f -> f.isDirectory() && RUN_DIRECTORY.matcher(f.getName()).matches()));
+  }
+
+  @Override
+  public NotificationDto process(File runDirectory, TimeZone tz) throws IOException {
+    // We create one DTO for a run, but there are going to be many wells with
+    // independent and
+    // duplicate metadata that will simply
+    // overwrite in the shared DTO. If the data differs, the last well wins.
+    PacBioNotificationDto dto = new PacBioNotificationDto();
+    dto.setPairedEndRun(false);
+    dto.setSequencerFolderPath(runDirectory.getAbsolutePath());
+
+    // Get the number of SMRT Cells in the run directory
+    dto.setLaneCount((int) getCellDirectory(runDirectory).count());
```

Comment:
getCellDirectory(runDirectory) is only being used in this file for the value of `.count()`. What do you think of saving the count to a variable up here and using the variable in lieu of the `getCellDirectory(runDirectory).count()` calls throughout?